Application/EdFi.Ods.Admin/Security/IdentityProvider.cs

```diff
@@ -0,0 +1,229 @@
+                {
+                    return false;
+                }
+
+                var result = identityUserManager.PasswordHasher.VerifyHashedPassword(identityUser.PasswordHash, password);
+
+                if (result == PasswordVerificationResult.Success)
+                {
+                    var authenticationManager = System.Web.HttpContext.Current.GetOwinContext().Authentication;
+
+                    var userIdentity = identityUserManager.CreateIdentity(
+                        identityUser, DefaultAuthenticationTypes.ApplicationCookie);
+
+                    authenticationManager.SignIn(new AuthenticationProperties() {IsPersistent = false}, userIdentity);
+                    return true;
+                }
+
+                return false;
+            }
+        }
+
+        public bool ResetUserPassword(string userName, string newPassword)
+        {
+            using (var identityUserManager = GetIdentityUserManager())
+            {
+                var identityUser = identityUserManager.FindByName(userName);
+
+                if (identityUser == null)
+                {
+                    return false;
+                }
+
+                string token = identityUserManager.GeneratePasswordResetToken(identityUser.Id);
+                var result = identityUserManager.ResetPassword(identityUser.Id, token, newPassword);
+
+                if (!result.Succeeded)
+                {
+                    _log.Error($"Failed resetting password for: {userName}. {string.Join(",", result.Errors)}");
+                }
+
+                return result.Succeeded;
+            }
+        }
+
+        public bool CreateUser(string userName, string email, string password, bool confirm = false)
+        {
+            using (var identityUserManager = GetIdentityUserManager())
+            {
+                IdentityResult result = identityUserManager.Create(
+                    new IdentityUser()
+                    {
+                        UserName = userName,
+                        Email = email,
+                        EmailConfirmed = confirm
+                    }, password);
+
+                if (!result.Succeeded)
+                {
+                    _log.Error($"Failed adding user: {userName} to asp net security. {string.Join(",", result.Errors)}");
+                }
+
+                return result.Succeeded;
+            }
+        }
+
+        public string GenerateEmailConfirmationToken(string email)
+        {
+            using (var identityUserManager = GetIdentityUserManager())
+            {
+                var identityUser = identityUserManager.FindByEmail(email);
+                return identityUserManager.GenerateEmailConfirmationToken(identityUser.Id);
+            }
+        }
+
+        public bool ConfirmEmailWithToken(string email, string token)
+        {
+            using (var identityUserManager = GetIdentityUserManager())
+            {
+                IdentityResult result = null;
```

Comment:
This is only used (and can be declared) inside the if (identityUser != null) check.